.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,841 @@
+      ##############################
+
+      - name: Set SECURE_BOOT_FEATURE_FLAG
+        if: inputs.deploy-hostapp == true
+        run: |
+          if [ -n "${{ inputs.sign-image }}" = "true" ]; then
+            echo "SECURE_BOOT_FEATURE_FLAG=yes" >> $GITHUB_ENV
+          else
+            echo "SECURE_BOOT_FEATURE_FLAG=no" >> $GITHUB_ENV
+          fi
+
+      - name: Pull helper image
+        id: hostapp-helper-image
+        if: inputs.deploy-hostapp == true
+        env:
+          HELPER_IMAGE_REPO: ghcr.io/balena-os/balena-yocto-scripts
+          YOCTO_SCRIPTS_VERSION: ${{ steps.balena-lib.outputs.yocto_scripts_version }}
+          YOCTO_SCRIPTS_REF: ${{ steps.balena-lib.outputs.yocto_scripts_ref }}
+          HELPER_IMAGE_VARIANT: balena-push-env
+        run: |
+          image_tag="${HELPER_IMAGE_REPO}:${YOCTO_SCRIPTS_VERSION}-${HELPER_IMAGE_VARIANT}"
+          if ! docker pull "${image_tag}"; then
+            image_tag="${HELPER_IMAGE_REPO}:${YOCTO_SCRIPTS_REF}-${HELPER_IMAGE_VARIANT}"
+            docker pull "${image_tag}"
+          fi
+
+          image_id="$(docker images --format "{{.ID}}" "${image_tag}")"
+          echo "id=${image_id}" >> $GITHUB_OUTPUT
+
+      - name: Check Balena CLI
+        run: |
+          # Test installation works
+          balena --version
+
+      ## SELECT WHICH API KEY TO USE BASED ON ENV
+      - name: Select Balena API key for Prod
+        if: inputs.environment == 'production'
+        run: echo "BALENAOS_TOKEN=${{ secrets.BALENA_API_KEY }}" >> $GITHUB_ENV
+
+      - name: Select Balena API key for staging
+        if: inputs.environment == 'staging'
+        run: echo "BALENAOS_TOKEN=${{ secrets.BALENAOS_STAGING_TOKEN }}" >> $GITHUB_ENV
+
+
+
+      # TODO: replace this with balena-io/deploy-to-balena-action when it supports deploy-only
+      # https://github.com/balena-io/deploy-to-balena-action/issues/286
+      - name: Deploy to balena
+        if: inputs.deploy-hostapp == true
+        env:
+          BALENAOS_ACCOUNT: ${{ vars.HOSTAPP_ORG || 'balena_os' }}
+          SLUG: "${{ steps.balena-lib.outputs.device_slug }}"
+          APPNAME: "${{ steps.balena-lib.outputs.device_slug }}"
+          #BALENAOS_TOKEN: "${{ secrets.BALENA_API_KEY }}"
+          META_BALENA_VERSION: "${{ steps.balena-lib.outputs.meta_balena_version }}"
+          RELEASE_VERSION: "${{ steps.balena-lib.outputs.os_version }}"
+          BOOTABLE: 1
+          TRANSLATION: "v6"
+          DEPLOY: yes
+          FINAL: no
+          ESR: "${{ inputs.deploy-esr }}"
+          balenaCloudEmail:
+          balenaCloudPassword:
+          #HOSTAPP_PATH: ${{ runner.temp }}/deploy/balena-image.docker
+          VERSION: ${{ steps.balena-lib.outputs.os_version }}
+        run: |
+          set -e
+
+          ## What is that for??
+          # if [ -f "/host/appimage.docker" ]; then
+          #   _local_image=$(docker load -i /host/appimage.docker | cut -d: -f1 --complement | tr -d " " )
+          # fi
```

Comment:
I don't recall all, but what I remember is that it was made to work for both containers and native, so if there was an `/host/appimage.docker` it means it's running in a container so it uses that image. Probably to do with how hostapp extensions were built at the time.